core/src/main/scala/org/json4s/Formats.scala

```diff
@@ -96,6 +96,14 @@ trait Formats extends Serializable { self: Formats =>
   def alwaysEscapeUnicode: Boolean = false
   def strictFieldDeserialization: Boolean = false
 
+  /**
+   * Setting to false preserves library's behavior prior to 3.6, where companion object constructors were only
+   * considered when deserializing if there were no primary constructors. Setting to true preserves the
+   * backwards-incompatible change made in 3.6 to always consider companion object constructors when deserializing
+   * (https://github.com/json4s/json4s/pull/487).
+   */
+  def alwaysConsiderCompanionConstructors: Boolean = true
```

Comment:
What does the `always` refer to here?
Shouldn't the setting be called `considerCompanionConstructors`?